sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/MicroBatchExecution.scala

```diff
@@ -590,7 +591,7 @@ class MicroBatchExecution(
     val newBatchesPlan = logicalPlan transform {
       // For v1 sources.
       case StreamingExecutionRelation(source, output, catalogTable) =>
-        newData.get(source).map { dataPlan =>
+        mutableNewData.get(source).map { dataPlan =>
```

Comment:
While we are here, probably less intrusive change would be moving (L594 ~ L610) to L567.

Reply:
Thanks, I initially thought about that, but we need to know the `output` from `StreamingExecutionRelation(source, output, catalogTable)` to resolve `_metadata` right (L591 ~ L593)?

Reply:
Yeah, you're right. I missed that.

Btw, looks like my change (tagging catalogTable into LogicalRelation) will also fall into this bug. Thanks for fixing this.

Reply:
Np - an unintentional fix :-)
Thanks for helping!

Reply:
Maybe, we may want to check the case of self-union / self-join to verify we really didn't break things. This works only when this condition is true `leaf : source = 1 : 1` (otherwise we are overwriting the value in map), while the code comment of ProgressReporter tells there are counter cases.

Reply:
Got it - could you share an example? In this case, does that mean the `leaf : source = 1 : N`?

Reply:
The code comment actually doesn't say much and I'm speculating. Let's just try a best effort, self-union and self-join. `df = spark.readStream... -> df.union(df)` / `df = spark.readStream... -> df.join(df)`